### PR TITLE
Use observables across Angular app

### PR DESCRIPTION
**Pull request by wmarques**
Made some improvements to leverage the reactive programming features of rxJs with Angular. 
I also cleaned up some useless asynchronous code (ie language Helper)

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/layouts/profiles/page-ribbon.component.ts.ejs [hunk 1]
```diff
@@ -17,27 +17,28 @@
  limitations under the License.
 -%>
 import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 import { ProfileService } from './profile.service';
 import { ProfileInfo } from './profile-info.model';
```
Comment (kaidohallik) [inline generators/client/templates/angular/src/main/webapp/app/layouts/profiles/page-ribbon.component.ts.ejs]: This import can be removed now

**Comment by wmarques** [comment]
@kaidohallik thanks for the review ! I'll apply your comments ASAP :)

**Comment by kaidohallik** [comment]
@wmarques, I'm interested moving on with this PR. If you have lack of time then maybe I can try to help if you are ok with this.  
Maybe try to do minimal additional changes here, necessary is to pass CI, other changes can maybe be follow ups?

**Comment by wmarques** [comment]
@kaidohallik I'll do that today or tomorrow that's ok sorry :)

**Comment by kaidohallik** [comment]
@wmarques, ok, thanks!

**Comment by kaidohallik** [comment]
@wmarques, I looked at failing test and found out:
* current code in master is wrong: https://github.com/jhipster/generator-jhipster/blob/master/generators/client/templates/angular/src/main/webapp/app/account/sessions/sessions.component.ts.ejs#L50-L62 because Angular `http` component routes response to `Observable` error chain if HTTP error occurs: https://angular.io/guide/http#getting-error-details
* in this PR you fixed this 🚀 
* but test is still wrong, it creates `Observable` into success chain, solution is: import `throwError` from `rxjs` and instead of
```
          of({
            status: 400
          })
```
use for example
```
          throwError({})
```

**Comment by wmarques** [comment]
@kaidohallik thanks ! Checking this :)

**Comment by wmarques** [comment]
LGTM, I'll do the other improvements suggested by @kaidohallik in a further PR if that's ok :)

@vishal423 if you have some time to review and merge this :)

File: generators/client/templates/angular/src/main/webapp/app/core/login/login.service.ts.ejs [hunk 2]
```diff
@@ -51,32 +46,11 @@ export class LoginService {
...
     }
     <%_ } else { _%>
-    login(credentials, callback?) {
-        const cb = callback || function() {};
-
-        return new Promise((resolve, reject) => {
-            this.authServerProvider.login(credentials).subscribe((data) => {
-                this.accountService.identity(true).then((account) => {
-                    <%_ if (websocket === 'spring-websocket') { _%>
-                    this.trackerService.sendActivity();
```
[the remark this answers is not in the document]
Comment (kaidohallik) [reply generators/client/templates/angular/src/main/webapp/app/core/login/login.service.ts.ejs]: In current code:
* `loginService.login` calls `this.accountService.identity(true)`
* `accountService.identity(true)` calls `this.trackerService.connect();`
* `trackerService.connect` calls `this.sendActivity();`

Previous version called `sendActivity()` twice on login, current code calls this only once.

**Comment by wmarques** [comment]
Thanks guys for the review and the comments, I'll handle this during the weekend !

File: generators/client/templates/angular/src/main/webapp/app/core/auth/account.service.ts.ejs [hunk 3]
```diff
@@ -32,10 +33,10 @@ import { <%=jhiPrefixCapitalized%>TrackerService } from '../tracker/tracker.serv
 
 @Injectable({providedIn: 'root'})
 export class AccountService  {
-    private userIdentity: any;
+    private userIdentity: Account;
```
[the remark this answers is not in the document]
Comment (wmarques) [reply generators/client/templates/angular/src/main/webapp/app/core/auth/account.service.ts.ejs]: I don't understand, you mean all the components should use authenticationState ?

File: generators/client/templates/angular/src/main/webapp/app/core/login/login.service.ts.ejs (hunk 2 above)
[the remark this answers is not in the document]
Comment (wmarques) [reply generators/client/templates/angular/src/main/webapp/app/core/login/login.service.ts.ejs]: Yep I think `this.trackerService.sendActivity();` wasn't necessary

File: generators/client/templates/angular/src/main/webapp/app/core/auth/account.service.ts.ejs (hunk 3 above)
[the remark this answers is not in the document]
Comment (wmarques) [reply generators/client/templates/angular/src/main/webapp/app/core/auth/account.service.ts.ejs]: I don't understand, you mean all the components should listen to `authenticationState` ?

[the remark this answers is not in the document]
Comment (vishal423) [reply generators/client/templates/angular/src/main/webapp/app/core/auth/account.service.ts.ejs]: I think that's fine for now..

**Comment by vishal423** [comment]
thanks @wmarques

**Pull request merged**